harness/tests/experiment/pytorch/test_pytorch_trial.py

```diff
@@ -67,110 +68,114 @@ def test_require_global_batch_size(self) -> None:
     def test_onevar_single(self) -> None:
         """Assert that the training loss and validation error decrease monotonically."""
 
-        def make_workloads() -> workload.Stream:
-            trainer = utils.TrainAndValidate()
+        class TrainingMetricsCallback(pytorch.PyTorchCallback):
```

Comment:
This might be a sign that there's more thought that should be put into the API design.

Local training mode doesn't seem very compelling if there's no way to get metrics out.

That doesn't necessarily mean it has to happen now, but we should probably think about how we _would_ return metrics, and make sure we could extend to support that in the future.

Reply:
decided not to ship local training mode for the time being.

Reply:
we decided emitting metrics is an obvious missing feature, but we're leaving it out of scope for the current work